eva-accession-release/src/test/java/uk/ac/ebi/eva/accession/release/io/VariantContextWriterTest.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.release.io;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.assertj.core.util.Sets;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import uk.ac.ebi.eva.accession.release.steps.processors.VariantToVariantContextProcessor;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class VariantContextWriterTest {
+
+    private static final String ID = "rs123";
+
+    private static final String CHR_1 = "1";
+
+    private static final String FILE_ID = "fileId";
+
+    private static final String SNP_SEQUENCE_ONTOLOGY = "SO:0001483";
+
+    private static final String DELETION_SEQUENCE_ONTOLOGY = "SO:0000159";
+
+    private static final String INSERTION_SEQUENCE_ONTOLOGY = "SO:0000667";
+
+    private static final String INDEL_SEQUENCE_ONTOLOGY = "SO:1000032";
+
+    private static final String TANDEM_REPEAT_SEQUENCE_ONTOLOGY = "SO:0000705";
+
+    private static final String SEQUENCE_ALTERATION_SEQUENCE_ONTOLOGY = "SO:0001059";
+
+    private static final String MNV_SEQUENCE_ONTOLOGY = "SO:0002007";
+
+    private static final String VARIANT_CLASS_KEY = "VC";
+
+    private static final String STUDY_ID_KEY = "SID";
+
+    private static final String STUDY_1 = "study_1";
+
+    private static final String STUDY_2 = "study_2";
+
+    private static final String REFERENCE_ASSEMBLY = "GCA_00000XXX.X";
+
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+    @Test
+    public void basicWrite() throws Exception {
+        File output = temporaryFolder.newFile("test.vcf");
```

Comment:
In order to avoid possible clashes / dependencies among tests, it would be preferable to use a different file per test case.